flycheck.el

```diff
@@ -6085,6 +6085,22 @@ objects)."
 
 
 ;;; Convenience definition of command-syntax checkers
+
+;; We need the byte-compiler to be aware of this function when macroexpanding
+;; `flycheck-define-checker'.
+(eval-and-compile
+  ;; We allow unquoted function symbols and lambdas as arguments to
+  ;; `flycheck-define-checker'.  When that macro is defined, like in this file,
+  ;; it works fine.  However, when wrapping `flycheck-define-checker' in a
+  ;; `with-eval-after-load flycheck', the body of the latter is macroexpanded
+  ;; *before* flycheck.el has been loaded, and thus before the macro is defined.
+  ;; Unquoted function symbols are still unquoted, but (lambda ...) forms are
+  ;; macroexpanded into (function (lambda ...)), and thus should *not* be
+  ;; quoted.  We use `flycheck-maybe-quote' to avoid double-quoting.
```

Comment:
I'd add a link to one of the Github issues and a note that we intend to get rid of this function after a while.